component/otelcol/internal/featuregate/install.go

```diff
@@ -19,7 +21,26 @@ func init() {
 }
 
 func enableFeatureGates(reg *featuregate.Registry) error {
+	// TODO(marctc): temporary workaround to fix issue with traces' metrics not
+	// being collected even flow is not enabled.
 	return reg.Apply(map[string]bool{
-		"telemetry.useOtelForInternalMetrics": true,
+		"telemetry.useOtelForInternalMetrics": isFlowRunning(),
 	})
 }
+
+func isFlowRunning() bool {
+	key, found := os.LookupEnv("AGENT_MODE")
+	if !found {
+		key, found := os.LookupEnv("EXPERIMENTAL_ENABLE_FLOW")
+		if !found {
+			return false
+		}
+		return key == "true" || key == "1"
```

Comment:
Should any of these values be case sensitive? Also, is it not better to get whether Flow is running from some kind of internal variable which is set on startup? The reason is that there would only be one place where we check the "EXPERIMENTAL_ENABLE_FLOW" and "AGENT_MODE" variables.

Reply:
I copied the code from here: https://github.com/grafana/agent/blob/main/cmd/agent/mode.go#L17 I thought reusing that code but is not possible as it's in `cmd/main`. A refactor would be needed but since it's a quickfix, I rather copy this bit for now.